torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -269,10 +269,10 @@ inline std::string retrieveDesyncReport(
 
 #ifdef USE_C10D_NCCL
 
-DebugInfoWriter::DebugInfoWriter(int rank) {
+DebugInfoWriter::DebugInfoWriter(int pgId, int rank) {
```

Comment:
i don't think we should need pgId in DebugInfoWriter.  The whole idea was the debug info is process-wide, so its not bound to a specific PG.  

If you're just using it here to log and ensure we only use pg 0 then i think thats ok.  but no need to pollute the filename with an extra 0_ IMO

Reply:
Well, actually during timeout, subPG will use this to dump the debug info and it will use local rank here. Let's say a local rank 0 (global rank 8) will write debug info when timeout in watchdog, then it will write into file name "*_rank_0" while for the global rank 0, it will also dump debug info to the file name "*_rank_0" and this is a race which will mess up our final dump files.

Reply:
For example in the log I see log files:

<img width="103" alt="image" src="https://github.com/pytorch/pytorch/assets/6937752/adcf68e5-9d3c-4bd8-8e3d-35f0af673092">

They are all from the same host, if we don't have uid in the file name, the file will all be the same.

Reply:
Hmm actually different sub-PG shared the same uid_, I will use global rank here instead.